server/models/device_history_instance.py

```
@@ -0,0 +1,216 @@
+        3. Apply limit/offset at the group level.
+        """
+        clauses, params = self._build_clauses(devGUID, changedColumn, changedBy)
+        where = f"WHERE {' AND '.join(clauses)}" if clauses else ""
+
+        rows = self._fetchall(
+            f"""
+            SELECT devGUID, timestamp, changedBy, changedColumn, oldValue, newValue
+            FROM DevicesHistory
+            {where}
+            ORDER BY timestamp DESC, changedBy, changedColumn
+            """,
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)